utils/src/main/kotlin/storage/FileArchiver.kt

```diff
@@ -65,6 +69,23 @@ class FileArchiver(
             }
         }
 
-        storage.write("${storagePath.removeSuffix("/")}/$ARCHIVE_FILE_NAME", zipFile.inputStream())
+        storage.write(getArchivePath(storagePath), zipFile.inputStream())
     }
+
+    fun unarchive(directory: File, storagePath: String): Boolean {
+        return try {
```

Comment:
Nit: No blocker, just chiming in that we usually make these "one-block" functions extension functions, even if this function is a bit longer in terms of lines of code.

Reply:
Done.